apps/frontend/src/libs/components/date-input/date-input.tsx

```diff
@@ -0,0 +1,109 @@
+import { Calendar } from "react-calendar";
+import {
+	type Control,
+	type FieldPath,
+	type FieldValues,
+} from "react-hook-form";
+
+import { Icon, Popover } from "~/libs/components/components.js";
+import { formatDate } from "~/libs/helpers/helpers.js";
+import {
+	useCallback,
+	useFormController,
+	usePopover,
+} from "~/libs/hooks/hooks.js";
+
+import { addDays, getDifferenceInDays } from "./libs/helpers/helpers.js";
+import { type DateValue } from "./libs/types/types.js";
+import styles from "./styles.module.css";
+
+type Properties<T extends FieldValues> = {
+	control: Control<T, null>;
+	maxRange?: number;
+	name: FieldPath<T>;
+};
+
+const DateInput = <T extends FieldValues>({
+	control,
+	maxRange,
+	name,
+}: Properties<T>): JSX.Element => {
+	const { field } = useFormController({
+		control,
+		name,
+	});
+
+	const { isOpened, onClose, onOpen } = usePopover();
+
+	const getSelectedDateRange = (): string => {
+		const [startDate, endDate] = field.value as Date[];
+
+		return `${formatDate(startDate as Date, "MMM d, yyyy")} - ${formatDate(endDate as Date, "MMM d, yyyy")}`;
+	};
+
+	const handleDateChange = useCallback(
+		(dates: DateValue): void => {
+			const [startDate, endDate] = dates as Date[];
+
+			if (maxRange && startDate && endDate) {
+				const diffInDays = getDifferenceInDays(endDate, startDate);
+
+				if (diffInDays > maxRange) {
+					const adjustedEndDate = addDays(startDate, maxRange);
+					field.onChange([startDate, adjustedEndDate]);
+
+					return;
+				}
+			}
```

Comment:
instead of this logic, can't we do something like this, and adjust maxDate instead?  Also is there a visual indication of the max date to be selected? for example, dates in the future are shown as disabled (grey)?

```tsx
const adjustedMaxDate = useMemo((): Date | undefined => {
  if(!startDate) {
    return maxDate;
  }

  const rangeMaxDate = addDays(startDate, maxRange);
  
  if(!maxDate) {
    return rangeMaxDate;
  }

  return maxDate?.getTime() > rangeMaxDate.getTime() ? rangeMaxDate : maxDate;
}, [startDate, maxDate, maxRange])
```

Reply:
Dates in future are not visible (as in the design) and we cannot navigate to months that are in future after max date.